addons/stock/models/stock_warehouse.py

```diff
@@ -233,8 +237,11 @@ def write(self, vals):
                     ('id', 'not in', picking_type_ids.ids),
                 ])
                 if picking_type_using_locations:
-                    raise UserError(_('%s use default source or destination locations from warehouse %s that will be archived.',
-                                    ', '.join(picking_type_using_locations.mapped('name')), warehouse.name))
+                    raise UserError(_(
+                        '%(type_list)s use default source or destination locations from warehouse %(warehouse)s that will be archived.',
+                        type_list=format_list(self.env, picking_type_using_locations.mapped('name')),
+                        warehouse=warehouse.name,
```

Comment:
I think this error message was maybe written at a time when it didn't block the transaction from being completed? So I guess it needs to be reworded to match. Logic may need to be fixed at some point to match previous behavior? :shrug: 

```suggestion
                        '%(operations)s have default source or destination locations within warehouse %(warehouse)s, therefore you cannot archive it.',
                        operations=format_list(self.env, picking_type_using_locations.mapped('name')),
                        warehouse=warehouse.name,
```